sql/2021/resource-hints/consoleLog_incorrect_type.sql

```diff
@@ -0,0 +1,34 @@
+#standardSQL
+# returns the number of pages which preload a resource of the incorrect script type
+
+# helper to create percent fields
+CREATE TEMP FUNCTION AS_PERCENT (freq FLOAT64, total FLOAT64) RETURNS FLOAT64 AS (
+  ROUND(SAFE_DIVIDE(freq, total), 4)
+);
+
+SELECT
+  client,
+  ARRAY_LENGTH(value) AS freq,
+  COUNT(0) AS total
+FROM (
+  SELECT
+    client,
+    REGEXP_EXTRACT_ALL(consoleLog, r'A preload for (.*?) is found, but is not used because the script type does not match.') AS value
```

Comment:
What does the `?` do in this regex?

Reply:
Non greedy, so if there are multiple matches it takes the shortest one.